tests/e2e/aws_disk_forensics.py

```diff
@@ -0,0 +1,315 @@
+  """End to end test of the AWS -> GCP disk copy workflow.
+
+  This end-to-end test runs directly on GCP and tests the following modules:
+    1. AWSVolumeSnapshotCollector
+    2. AWSSnapshotS3CopyCollector
+    3. S3ToGCSCopy
+    4. GCSToGCEImage
+    5. GCEDiskFromImage
+
+  To run this test, add your project information to a project_info.json file:
+
+  {
+    "gcp_project_id": "xxx", # required
+    "gcp_bucket": "xxx", # required
+    "gcp_zone": "xxx", # required
+    "aws_volume": "vol-xxx", # required
+    "aws_region": "xxx", # required
+    "aws_bucket": "xxx", # required
+    "aws_subnet": "subnet-xxx" # Required if @aws_region has no default subnet
+  }
+
+  Export a PROJECT_INFO environment variable with the absolute path to your
+  file: "user@terminal:~$ export PROJECT_INFO='absolute/path/project_info.json'"
+
+  You also need to already be authenticated to AWS in some way. ENVVARS are
+  recommended but otherwise see:
+  https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-configure.html
+  """
+
+  def __init__(self, *args, **kwargs):
+    super(AWSToGCPForensicsEndToEndTest, self).__init__(*args, **kwargs)
+    try:
+      project_info = ReadProjectInfo()
+    except (OSError, RuntimeError, ValueError) as exception:
+      self.error_msg = str(exception)
+      return
+
+    self.gcp_project_id = project_info['gcp_project_id']
+    self.gcp_bucket = project_info['gcp_bucket']
+    self.gcp_zone = project_info['gcp_zone']
+    self.aws_volume = project_info['aws_volume']
+    self.aws_region = project_info['aws_region']
+    self.aws_bucket = project_info['aws_bucket']
+    self.aws_subnet = project_info['aws_subnet']
+
+  def setUp(self):
+    self.incident_id = 'fake-incident-id'
+
+    self.test_state = state.DFTimewolfState(config.Config)
+    self._recipe = resources.Recipe("E2E Test Recipe", RECIPE, [])
+    self._recipes_manager = recipes_manager.RecipesManager()
+    self._recipes_manager.RegisterRecipe(self._recipe)
+
+  def testRunRecipe(self):
+    """Tests the recipe for AWS->GCP disk copy."""
+    warnings.filterwarnings(
+        action="ignore", message="unclosed", category=ResourceWarning)
+
+    # Load the recipe, set the arguments, and run
+    self.test_state.LoadRecipe(RECIPE, TEST_MODULES)
+    self.test_state.command_line_options = {
+      'aws_region': self.aws_region,
+      'gcp_zone': self.gcp_zone,
+      'volumes': self.aws_volume,
+      'aws_bucket': self.aws_bucket,
+      'gcp_bucket': self.gcp_bucket,
+      'subnet': self.aws_subnet,
+      'gcp_project': self.gcp_project_id
+    }
+    self.test_state.SetupModules()
+    self.test_state.RunModules()
+
+    # AWS Volume in count should equal GCE Disk out count, and be at least 1
+    self.assertGreaterEqual(
+        len(self.test_state.GetContainers(containers.AWSVolume)), 1)
+    self.assertEqual(len(self.test_state.GetContainers(containers.AWSVolume)),
+        len(self.test_state.GetContainers(containers.GCEDiskEvidence)))
```

Comment:
We're testing the internal state here, in e2e tests I think it would probably be better to leverage LCF to get the actual state of the remote project (and compare names that are there with the ones that dftimewolf claims to have created)

Reply:
Unless I'm misunderstanding what you're asking, that's what's happening here. 199->202 is making sure that we have the same number of input disks in AWS as outputs in GCE according to the internal state (and also that it's not zero), then 204->207 is making sure those GCE disks actually exist via LCF.

Reply:
Yup I get this - what I'm suggesting is that we rely less on the internal state (given that this is e2e) and interact with the remote cloud API to confirm that the disks have been created and have the expected characteristics.

We do this with the [gcp_disk_forensics tests](https://github.com/log2timeline/dftimewolf/blob/main/tests/e2e/gcp_disk_forensics.py#L183)

Reply:
Updated, as per OOB conversation.